New error on working workflow

**How would you describe the issue?**

From this weekend I have this error:

```
Automatic merges enabled for GitHub login: dependabot-preview[bot].
Found pull request information: {"reviewEdges":[{"node":{"state":"APPROVED"}}]}.
An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.
##[error]An unexpected error occurred: Required status check "Continuous integration" is in progress., GraphqlError: Required status check "Continuous integration" is in progress.
    at key (/home/runner/work/_actions/ridedott/dependabot-auto-merge-action/master/node_modules/@octokit/graphql/dist-node/index.js:42:1)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at approveAndMergePullRequestMutation (/home/runner/work/_actions/ridedott/dependabot-auto-merge-action/master/src/common/merge.ts:40:9)
    at Object.mergeWithRetry (/home/runner/work/_actions/ridedott/dependabot-auto-merge-action/master/src/common/merge.ts:49:14)
    at Object.logInfo (/home/runner/work/_actions/ridedott/dependabot-auto-merge-action/master/src/eventHandlers/pullRequest/index.ts:86:5).
```

**How can we reproduce the issue?**

Run this workflow:
https://github.com/mapfish/mapfish-print/blob/master/.github/workflows/dependabot-auto-merge.yaml

In // of a required and running workflow

**What are the expected results?**

Validate the pull request

**What are the actual results?**

Error in the workflow, e.-g.:

https://github.com/mapfish/mapfish-print/pull/1453/checks?check_run_id=1020852174

**How much does it hurt?**

It makes this action unusable for us...

--- a/cSpell.json
+++ b/cSpell.json
@@ -40,6 +40,7 @@
     "prettierrc",
     "rebased",
     "releaserc",
+    "retryable",
     "ridedott",
     "ridedottmerge",
     "webhook"

--- a/src/common/merge.ts
+++ b/src/common/merge.ts
@@ -1,3 +1,5 @@
+// cspell:ignore retryable
+
 import { getOctokit } from '@actions/github';
 
 import {
@@ -43,8 +45,23 @@ const merge = async (
   await octokit.graphql(mutation, { commitHeadline, pullRequestId });
 };
 
-const shouldRetry = (error: Error): boolean =>
-  error.message.includes('Base branch was modified.');
+const shouldRetry = (
+  error: Error,
+  retryCount: number,
+  maximumRetries: number,
+): boolean => {
+  const isRetryableError = error.message.includes('Base branch was modified.');
+
+  if (isRetryableError && retryCount > maximumRetries) {
+    logInfo(
+      `Unable to merge after ${retryCount.toString()} attempts. Retries exhausted.`,
+    );
+
+    return false;
+  }
+
+  return isRetryableError;
+};
 
 export const mergeWithRetry = async (
   octokit: ReturnType<typeof getOctokit>,
@@ -58,6 +75,20 @@ export const mergeWithRetry = async (
   try {
     await merge(octokit, details);
   } catch (error) {
+    if (shouldRetry(error, retryCount, maximumRetries)) {
+      const nextRetryIn = retryCount ** EXPONENTIAL_BACKOFF * MINIMUM_WAIT_TIME;
+
+      logInfo(`Retrying in ${nextRetryIn.toString()}...`);
+
+      await delay(nextRetryIn);
+
+      return await mergeWithRetry(octokit, {
+        ...details,
+        maximumRetries,
+        retryCount: retryCount + 1,
+      });
+    }
+
     logInfo(
       'An error ocurred while merging the Pull Request. This is usually ' +
         'caused by the base branch being out of sync with the target ' +
@@ -66,27 +97,5 @@ export const mergeWithRetry = async (
     );
     /* eslint-disable-next-line @typescript-eslint/no-base-to-string */
     logDebug(`Original error: ${(error as Error).toString()}.`);
-
-    if (shouldRetry(error) === false) {
-      return;
-    }
-
-    if (retryCount <= maximumRetries) {
-      const nextRetryIn = retryCount ** EXPONENTIAL_BACKOFF * MINIMUM_WAIT_TIME;
-
-      logInfo(`Retrying in ${nextRetryIn.toString()}...`);
-
-      await delay(nextRetryIn);
-
-      await mergeWithRetry(octokit, {
-        ...details,
-        maximumRetries,
-        retryCount: retryCount + 1,
-      });
-    } else {
-      logInfo(
-        `Unable to merge after ${retryCount.toString()} attempts. Retries exhausted.`,
-      );
-    }
   }
 };

--- a/src/common/merge.ts
+++ b/src/common/merge.ts
@@ -67,7 +67,11 @@ export const mergeWithRetry = async (
     /* eslint-disable-next-line @typescript-eslint/no-base-to-string */
     logDebug(`Original error: ${(error as Error).toString()}.`);
 
-    if (shouldRetry(error) && retryCount <= maximumRetries) {
+    if (shouldRetry(error) === false) {
+      return;
+    }
+
+    if (retryCount <= maximumRetries) {
       const nextRetryIn = retryCount ** EXPONENTIAL_BACKOFF * MINIMUM_WAIT_TIME;
 
       logInfo(`Retrying in ${nextRetryIn.toString()}...`);
@@ -80,7 +84,9 @@ export const mergeWithRetry = async (
         retryCount: retryCount + 1,
       });
     } else {
-      throw error;
+      logInfo(
+        `Unable to merge after ${retryCount.toString()} attempts. Retries exhausted.`,
+      );
     }
   }
 };

--- a/src/common/merge.ts
+++ b/src/common/merge.ts
@@ -1,5 +1,3 @@
-// cspell:ignore retryable
-
 import { getOctokit } from '@actions/github';
 
 import {

--- a/src/eventHandlers/checkSuite/index.spec.ts
+++ b/src/eventHandlers/checkSuite/index.spec.ts
@@ -395,7 +395,7 @@ describe('check Suite event handler', (): void => {
     );
     expect(infoSpy).toHaveBeenCalledWith('Retrying in 1000...');
     expect(infoSpy).toHaveBeenCalledWith('Retrying in 4000...');
-    expect(debugSpy).toHaveBeenCalledTimes(3);
+    expect(debugSpy).toHaveBeenCalledTimes(1);
     expect(debugSpy).toHaveBeenCalledWith(
       'Original error: HttpError: ##[error]GraphqlError: Base branch was modified. Review and try the merge again..',
     );

--- a/src/eventHandlers/checkSuite/index.spec.ts
+++ b/src/eventHandlers/checkSuite/index.spec.ts
@@ -10,7 +10,6 @@ import * as nock from 'nock';
 import { useSetTimeoutImmediateInvocation } from '../../../test/utilities';
 import { mergePullRequestMutation } from '../../graphql/mutations';
 import { AllowedMergeMethods } from '../../utilities/inputParsers';
-import * as log from '../../utilities/log';
 import { checkSuiteHandle } from '.';
 
 /* cspell:disable-next-line */
@@ -20,6 +19,7 @@ const COMMIT_HEADLINE = 'Update test';
 const octokit = getOctokit('SECRET_GITHUB_TOKEN');
 const infoSpy = jest.spyOn(core, 'info').mockImplementation();
 const warningSpy = jest.spyOn(core, 'warning').mockImplementation();
+const debugSpy = jest.spyOn(core, 'debug').mockImplementation();
 const getInputSpy = jest.spyOn(core, 'getInput').mockImplementation();
 
 beforeEach((): void => {
@@ -342,7 +342,7 @@ describe('check Suite event handler', (): void => {
   });
 
   it('retries up to two times before failing', async (): Promise<void> => {
-    expect.assertions(6);
+    expect.assertions(5);
 
     nock('https://api.github.com')
       .post('/graphql')
@@ -386,31 +386,25 @@ describe('check Suite event handler', (): void => {
         '##[error]GraphqlError: Base branch was modified. Review and try the merge again.',
       );
 
-    const logDebugSpy = jest.spyOn(log, 'logDebug');
-    const logInfoSpy = jest.spyOn(log, 'logInfo');
-
     useSetTimeoutImmediateInvocation();
 
-    try {
-      await checkSuiteHandle(octokit, 'dependabot-preview[bot]', 2);
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-      expect(error.message).toStrictEqual(
-        '##[error]GraphqlError: Base branch was modified. Review and try the merge again.',
-      );
-      expect(logDebugSpy).toHaveBeenCalledTimes(3);
-      expect(logInfoSpy.mock.calls[1][0]).toStrictEqual(
-        'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
-      );
-      expect(logInfoSpy.mock.calls[2][0]).toStrictEqual('Retrying in 1000...');
-      expect(logInfoSpy.mock.calls[4][0]).toStrictEqual('Retrying in 4000...');
-    }
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]', 2);
+
+    expect(infoSpy).toHaveBeenCalledWith(
+      'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
+    );
+    expect(infoSpy).toHaveBeenCalledWith('Retrying in 1000...');
+    expect(infoSpy).toHaveBeenCalledWith('Retrying in 4000...');
+    expect(debugSpy).toHaveBeenCalledTimes(3);
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Original error: HttpError: ##[error]GraphqlError: Base branch was modified. Review and try the merge again..',
+    );
   });
 
   it('fails the backoff strategy when the error is not "Base branch was modified"', async (): Promise<
     void
   > => {
-    expect.assertions(3);
+    expect.assertions(2);
 
     nock('https://api.github.com')
       .post('/graphql')
@@ -450,18 +444,13 @@ describe('check Suite event handler', (): void => {
       .post('/graphql')
       .reply(403, '##[error]GraphqlError: This is a different error.');
 
-    const logInfoSpy = jest.spyOn(log, 'logInfo');
+    await checkSuiteHandle(octokit, 'dependabot-preview[bot]', 2);
 
-    try {
-      await checkSuiteHandle(octokit, 'dependabot-preview[bot]', 2);
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-      expect(error.message).toStrictEqual(
-        '##[error]GraphqlError: This is a different error.',
-      );
-      expect(logInfoSpy.mock.calls[1][0]).toStrictEqual(
-        'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
-      );
-    }
+    expect(infoSpy).toHaveBeenCalledWith(
+      'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
+    );
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Original error: HttpError: ##[error]GraphqlError: This is a different error..',
+    );
   });
 });

--- a/src/eventHandlers/pullRequest/index.spec.ts
+++ b/src/eventHandlers/pullRequest/index.spec.ts
@@ -10,7 +10,6 @@ import * as nock from 'nock';
 import { useSetTimeoutImmediateInvocation } from '../../../test/utilities';
 import { mergePullRequestMutation } from '../../graphql/mutations';
 import { AllowedMergeMethods } from '../../utilities/inputParsers';
-import * as log from '../../utilities/log';
 import { pullRequestHandle } from '.';
 
 /* cspell:disable-next-line */
@@ -18,7 +17,9 @@ const PULL_REQUEST_ID = 'MDExOlB1bGxSZXF1ZXN0MzE3MDI5MjU4';
 const COMMIT_HEADLINE = 'Update test';
 
 const octokit = getOctokit('SECRET_GITHUB_TOKEN');
+const infoSpy = jest.spyOn(core, 'info').mockImplementation();
 const warningSpy = jest.spyOn(core, 'warning').mockImplementation();
+const debugSpy = jest.spyOn(core, 'debug').mockImplementation();
 const getInputSpy = jest.spyOn(core, 'getInput').mockImplementation();
 
 jest.spyOn(core, 'info').mockImplementation();
@@ -135,7 +136,7 @@ describe('pull request event handler', (): void => {
   });
 
   it('retries up to two times before failing', async (): Promise<void> => {
-    expect.assertions(6);
+    expect.assertions(5);
 
     nock('https://api.github.com')
       .post('/graphql')
@@ -178,31 +179,25 @@ describe('pull request event handler', (): void => {
         '##[error]GraphqlError: Base branch was modified. Review and try the merge again.',
       );
 
-    const logDebugSpy = jest.spyOn(log, 'logDebug');
-    const logInfoSpy = jest.spyOn(log, 'logInfo');
-
     useSetTimeoutImmediateInvocation();
 
-    try {
-      await pullRequestHandle(octokit, 'dependabot-preview[bot]', 2);
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-      expect(error.message).toStrictEqual(
-        '##[error]GraphqlError: Base branch was modified. Review and try the merge again.',
-      );
-      expect(logDebugSpy).toHaveBeenCalledTimes(3);
-      expect(logInfoSpy.mock.calls[1][0]).toStrictEqual(
-        'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
-      );
-      expect(logInfoSpy.mock.calls[2][0]).toStrictEqual('Retrying in 1000...');
-      expect(logInfoSpy.mock.calls[4][0]).toStrictEqual('Retrying in 4000...');
-    }
+    await pullRequestHandle(octokit, 'dependabot-preview[bot]', 2);
+
+    expect(infoSpy).toHaveBeenCalledWith(
+      'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
+    );
+    expect(infoSpy).toHaveBeenCalledWith('Retrying in 1000...');
+    expect(infoSpy).toHaveBeenCalledWith('Retrying in 4000...');
+    expect(debugSpy).toHaveBeenCalledTimes(3);
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Original error: HttpError: ##[error]GraphqlError: Base branch was modified. Review and try the merge again..',
+    );
   });
 
   it('fails the backoff strategy when the error is not "Base branch was modified"', async (): Promise<
     void
   > => {
-    expect.assertions(3);
+    expect.assertions(2);
 
     nock('https://api.github.com')
       .post('/graphql')
@@ -241,18 +236,13 @@ describe('pull request event handler', (): void => {
       .post('/graphql')
       .reply(403, '##[error]GraphqlError: This is a different error.');
 
-    const logInfoSpy = jest.spyOn(log, 'logInfo');
+    await pullRequestHandle(octokit, 'dependabot-preview[bot]', 2);
 
-    try {
-      await pullRequestHandle(octokit, 'dependabot-preview[bot]', 2);
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-      expect(error.message).toStrictEqual(
-        '##[error]GraphqlError: This is a different error.',
-      );
-      expect(logInfoSpy.mock.calls[1][0]).toStrictEqual(
-        'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
-      );
-    }
+    expect(infoSpy).toHaveBeenCalledWith(
+      'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
+    );
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Original error: HttpError: ##[error]GraphqlError: This is a different error..',
+    );
   });
 });

--- a/src/eventHandlers/pullRequest/index.spec.ts
+++ b/src/eventHandlers/pullRequest/index.spec.ts
@@ -188,7 +188,7 @@ describe('pull request event handler', (): void => {
     );
     expect(infoSpy).toHaveBeenCalledWith('Retrying in 1000...');
     expect(infoSpy).toHaveBeenCalledWith('Retrying in 4000...');
-    expect(debugSpy).toHaveBeenCalledTimes(3);
+    expect(debugSpy).toHaveBeenCalledTimes(1);
     expect(debugSpy).toHaveBeenCalledWith(
       'Original error: HttpError: ##[error]GraphqlError: Base branch was modified. Review and try the merge again..',
     );

--- a/src/eventHandlers/push/index.spec.ts
+++ b/src/eventHandlers/push/index.spec.ts
@@ -10,7 +10,6 @@ import * as nock from 'nock';
 import { useSetTimeoutImmediateInvocation } from '../../../test/utilities';
 import { mergePullRequestMutation } from '../../graphql/mutations';
 import { AllowedMergeMethods } from '../../utilities/inputParsers';
-import * as log from '../../utilities/log';
 import { pushHandle } from '.';
 
 /* cspell:disable-next-line */
@@ -20,6 +19,7 @@ const COMMIT_HEADLINE = 'Update test';
 const octokit = getOctokit('SECRET_GITHUB_TOKEN');
 const infoSpy = jest.spyOn(core, 'info').mockImplementation();
 const warningSpy = jest.spyOn(core, 'warning').mockImplementation();
+const debugSpy = jest.spyOn(core, 'debug').mockImplementation();
 const getInputSpy = jest.spyOn(core, 'getInput').mockImplementation();
 
 beforeEach((): void => {
@@ -262,7 +262,7 @@ describe('push event handler', (): void => {
   });
 
   it('retries up to two times before failing', async (): Promise<void> => {
-    expect.assertions(6);
+    expect.assertions(5);
 
     nock('https://api.github.com')
       .post('/graphql')
@@ -298,31 +298,25 @@ describe('push event handler', (): void => {
         '##[error]GraphqlError: Base branch was modified. Review and try the merge again.',
       );
 
-    const logDebugSpy = jest.spyOn(log, 'logDebug');
-    const logInfoSpy = jest.spyOn(log, 'logInfo');
-
     useSetTimeoutImmediateInvocation();
 
-    try {
-      await pushHandle(octokit, 'dependabot-preview[bot]', 2);
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-      expect(error.message).toStrictEqual(
-        '##[error]GraphqlError: Base branch was modified. Review and try the merge again.',
-      );
-      expect(logDebugSpy).toHaveBeenCalledTimes(3);
-      expect(logInfoSpy.mock.calls[1][0]).toStrictEqual(
-        'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
-      );
-      expect(logInfoSpy.mock.calls[2][0]).toStrictEqual('Retrying in 1000...');
-      expect(logInfoSpy.mock.calls[4][0]).toStrictEqual('Retrying in 4000...');
-    }
+    await pushHandle(octokit, 'dependabot-preview[bot]', 2);
+
+    expect(infoSpy).toHaveBeenCalledWith(
+      'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
+    );
+    expect(infoSpy).toHaveBeenCalledWith('Retrying in 1000...');
+    expect(infoSpy).toHaveBeenCalledWith('Retrying in 4000...');
+    expect(debugSpy).toHaveBeenCalledTimes(3);
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Original error: HttpError: ##[error]GraphqlError: Base branch was modified. Review and try the merge again..',
+    );
   });
 
   it('fails the backoff strategy when the error is not "Base branch was modified"', async (): Promise<
     void
   > => {
-    expect.assertions(3);
+    expect.assertions(2);
 
     nock('https://api.github.com')
       .post('/graphql')
@@ -354,18 +348,13 @@ describe('push event handler', (): void => {
       .post('/graphql')
       .reply(403, '##[error]GraphqlError: This is a different error.');
 
-    const logInfoSpy = jest.spyOn(log, 'logInfo');
+    await pushHandle(octokit, 'dependabot-preview[bot]', 2);
 
-    try {
-      await pushHandle(octokit, 'dependabot-preview[bot]', 2);
-    } catch (error) {
-      expect(error).toBeInstanceOf(Error);
-      expect(error.message).toStrictEqual(
-        '##[error]GraphqlError: This is a different error.',
-      );
-      expect(logInfoSpy.mock.calls[1][0]).toStrictEqual(
-        'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
-      );
-    }
+    expect(infoSpy).toHaveBeenCalledWith(
+      'An error ocurred while merging the Pull Request. This is usually caused by the base branch being out of sync with the target branch. In this case, the base branch must be rebased. Some tools, such as Dependabot, do that automatically.',
+    );
+    expect(debugSpy).toHaveBeenCalledWith(
+      'Original error: HttpError: ##[error]GraphqlError: This is a different error..',
+    );
   });
 });

--- a/src/eventHandlers/push/index.spec.ts
+++ b/src/eventHandlers/push/index.spec.ts
@@ -307,7 +307,7 @@ describe('push event handler', (): void => {
     );
     expect(infoSpy).toHaveBeenCalledWith('Retrying in 1000...');
     expect(infoSpy).toHaveBeenCalledWith('Retrying in 4000...');
-    expect(debugSpy).toHaveBeenCalledTimes(3);
+    expect(debugSpy).toHaveBeenCalledTimes(1);
     expect(debugSpy).toHaveBeenCalledWith(
       'Original error: HttpError: ##[error]GraphqlError: Base branch was modified. Review and try the merge again..',
     );

